src/hotspot/share/gc/shared/cardTable.hpp

```diff
@@ -49,47 +49,21 @@ class CardTable: public CHeapObj<mtGC> {
   CardValue*      _byte_map;         // the card marking array
   CardValue*      _byte_map_base;
 
-  int _cur_covered_regions;
+  // Some barrier sets create tables whose elements correspond to parts of
+  // the heap; the CardTableBarrierSet is an example.  Such barrier sets will
+  // normally reserve space for such tables, and commit parts of the table
+  // "covering" parts of the heap that are committed. At most one covered
+  // region per generation is needed.
+  static constexpr int max_covered_regions = 2;
 
   // The covered regions should be in address order.
   MemRegion* _covered;
-  // The committed regions correspond one-to-one to the covered regions.
-  // They represent the card-table memory that has been committed to service
-  // the corresponding covered region.  It may be that committed region for
-  // one covered region corresponds to a larger region because of page-size
-  // roundings.  Thus, a committed region for one covered region may
-  // actually extend onto the card-table space for the next covered region.
-  MemRegion* _committed;
 
   // The last card is a guard card; never committed.
   MemRegion _guard_region;
```

Comment:
@albertnetymk : 

It looks like, following this refactor, you have stopped using `guard_region` for its previous role. I'd either put some of those checks back in, or just delete this now otherwise obsolete field.

It is possible, however, that I am missing something here.

Thanks!

Reply:
Doh, scratch that comment.